### PR TITLE
Provide EnumFactory

### PR DESCRIPTION
**Pull request by KochTobi**
This PR provides the EnumFactory that when implemented for an Enum with String values provides the reverse mapping from String to Enum.KEY

### CODE REVIEW

**Review by Steffengreiner — DISMISSED** [review]
LGTM!

File: src/test/groovy/life/qbic/datamodel/dtos/business/AcademicTitleFactorySpec.groovy [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package life.qbic.datamodel.dtos.business
+
+import spock.lang.Specification
+
+/**
+ * <short description>
+ *
+ * <detailed description>
+ *
```
Comment (jenniferboedker) [inline src/test/groovy/life/qbic/datamodel/dtos/business/AcademicTitleFactorySpec.groovy:9]: NIT: either describe or delete it ;)

Comment (KochTobi) [reply src/test/groovy/life/qbic/datamodel/dtos/business/AcademicTitleFactorySpec.groovy]: a4572f6

**Review by sven1103 — APPROVED** [review]
LGTM, nice work

**Pull request merged**